src/setuptools_scm/__init__.py

```diff
@@ -43,27 +43,35 @@
 
 def dump_version(
     root: _t.PathT,
-    version: str,
+    version: Version,
```

Comment:
the argument rename is a breaking change change would silently break downstream users
as such